bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowerMaxCommDriver.java

```diff
@@ -0,0 +1,621 @@
+     *
+     * @return true if connected or false if not
+     */
+    public boolean close() {
+        if (connector != null) {
+            connector.close();
+        }
+        downloadRunning = false;
+        return isConnected();
+    }
+
+    /**
+     * @return true if connected to the PowerMax alarm system or false if not
+     */
+    public boolean isConnected() {
+        return (connector != null) && connector.isConnected();
+    }
+
+    /**
+     * @return the last message sent to the PowerMax alarm system
+     */
+    public synchronized PowerMaxBaseMessage getLastSendMsg() {
+        return lastSendMsg;
+    }
+
+    /**
+     * @return the time in milliseconds used to set time and date
+     */
+    public Long getSyncTimeCheck() {
+        return syncTimeCheck;
+    }
+
+    /**
+     * Compute the CRC of a message
+     *
+     * @param data
+     *            the buffer containing the message
+     * @param len
+     *            the size of the message in the buffer
+     *
+     * @return the computed CRC
+     */
+    public static byte computeCRC(byte[] data, int len) {
+        long checksum = 0;
+        for (int i = 1; i < (len - 2); i++) {
+            checksum = checksum + (data[i] & 0x000000FF);
+        }
+        checksum = 0xFF - (checksum % 0xFF);
+        if (checksum == 0xFF) {
+            checksum = 0;
+        }
+        return (byte) checksum;
+    }
+
+    /**
+     * Send an ACK for a received message
+     *
+     * @param msg
+     *            the received message object
+     * @param ackType
+     *            the type of ACK to be sent
+     *
+     * @return true if the ACK was sent or false if not
+     */
+    public synchronized boolean sendAck(PowerMaxBaseMessage msg, byte ackType) {
+        int code = msg.getCode();
+        byte[] rawData = msg.getRawData();
+        byte[] ackData;
+        if ((code >= 0x80) || ((code < 0x10) && (rawData[rawData.length - 3] == 0x43))) {
+            ackData = new byte[] { 0x0D, ackType, 0x43, 0x00, 0x0A };
+        } else {
+            ackData = new byte[] { 0x0D, ackType, 0x00, 0x0A };
+        }
+
+        logger.debug("sendAck(): sending message {}", DatatypeConverter.printHexBinary(ackData));
+        boolean done = sendMessage(ackData);
+        if (!done) {
+            logger.debug("sendAck(): failed");
+        }
+        return done;
+    }
+
+    /**
+     * Send a message to the PowerMax alarm panel to change arm mode
+     *
+     * @param armMode
+     *            the arm mode. Allowed values are: Disarmed, Stay, Armed,
+     *            StayInstant, ArmedInstant, Night, NightInstant
+     * @param pinCode
+     *            the PIN code. A string of 4 characters is expected
+     *
+     * @return true if the message was sent or false if not
+     */
+    public boolean requestArmMode(String armMode, String pinCode) {
+        logger.debug("requestArmMode(): armMode = {}", armMode == null ? "null" : armMode);
```

Comment:
I thought that if `armMode` is `null`, the logger class would do the same as your ternary operator?